TrafficLight.py

```diff
@@ -0,0 +1,111 @@
+""" Represents the state of a traffic flow at a single light, a matrix made of rows that each
+represent a state of the traffic system. Within each row are floats between 0 and 1,
+representing the relative speed of traffic that can be processed in the column's direction to
+the direction of the row, IE a straight passage might be 1,
+a right turn might be 0.25, and a left turn might be 0.1""
+
+Also takes in a list of names exactly equal to the number of directions, with the nth name referring
+to activities within the nth row and column [name_list]
+Also takes in a list of timers with the length of each state listed in seconds (can be floats) [timers]
+"""
+
+
+class TrafficLight:
+    def __init__(self, name, traffic_matrix, traffic_flow_matrix_list, name_list, timers):
+        self.name = name
+        self.traffic_matrix = traffic_matrix
```

Comment:
For naming, try not to include the type of the variables into the name. For example, use `self.traffic` instead of `self.traffic_matrix`, and use `self.name` instead of `self.name_list`. When the typing is included in the function header (refer below), it would not be necessary to include them in the variable names.

Reply:
that makes a lot of sense. Great style tip, thanks